src/software/geom/polygon.h

```diff
@@ -43,6 +43,13 @@ class Polygon : public virtual Shape
      */
     Polygon expand(double expansion_amount) const;
 
+    /**
+     * Returns perimeter length of the polygon
+     *
+     * @return perimeter length of the polygon
+     */
+    double perimeter() const;
```

Comment:
can we add 1 or 2 tests for this as well?

Reply:
Ok